lib/sinon/fake.js

```diff
@@ -37,70 +37,87 @@ function wrapFunc(f) {
     return proxy;
 }
 
-function fake(f) {
-    if (arguments.length > 0 && typeof f !== "function") {
-        throw new TypeError("Expected f argument to be a Function");
-    }
+function fakeClass() {
+    var promiseLib;
 
-    return wrapFunc(f);
-}
+    function fake(f) {
+        if (arguments.length > 0 && typeof f !== "function") {
+            throw new TypeError("Expected f argument to be a Function");
+        }
 
-fake.returns = function returns(value) {
-    function f() {
-        return value;
+        return wrapFunc(f);
     }
 
-    return wrapFunc(f);
-};
-
-fake.throws = function throws(value) {
-    function f() {
-        throw getError(value);
-    }
+    fake.returns = function returns(value) {
+        function f() {
+            return value;
+        }
 
-    return wrapFunc(f);
-};
+        return wrapFunc(f);
+    };
 
-fake.resolves = function resolves(value) {
-    function f() {
-        return Promise.resolve(value);
-    }
+    fake.throws = function throws(value) {
+        function f() {
+            throw getError(value);
+        }
 
-    return wrapFunc(f);
-};
+        return wrapFunc(f);
+    };
 
-fake.rejects = function rejects(value) {
-    function f() {
-        return Promise.reject(getError(value));
-    }
+    fake.resolves = function resolves(value) {
+        function f() {
+            if (promiseLib) {
+                return promiseLib.resolve(value);
+            }
+            return Promise.resolve(value);
+        }
 
-    return wrapFunc(f);
-};
+        return wrapFunc(f);
+    };
 
-function yieldInternal(async, values) {
-    function f() {
-        var callback = arguments[arguments.length - 1];
-        if (typeof callback !== "function") {
-            throw new TypeError("Expected last argument to be a function");
+    fake.rejects = function rejects(value) {
+        function f() {
+            if (promiseLib) {
+                return promiseLib.reject(value);
+            }
```

Comment:
```suggestion
```

Reply:
Hi @fatso83, 

That actually makes a whole lot of sense, will take those edits. Thanks!

Reply:
Also noticed that i'm not wrapping the the rejected object as an Error. Will correct that as well.